pkg/services/govmomi/util.go

```diff
@@ -58,46 +58,80 @@ func sanitizeIPAddrs(ctx *context.VMContext, ipAddrs []string) []string {
 //   3. If it is not found by instance UUID, fallback to an inventory path search
 //      using the vm folder path and the VSphereVM name
 func findVM(ctx *context.VMContext) (types.ManagedObjectReference, error) {
+	var err error
+
+	moRef, biosUUIDMatch, err := findVMWithBiosUUID(ctx)
+	if biosUUIDMatch {
+		return moRef, err
+	}
+
+	moRef, err = findVMWithInstanceUUID(ctx)
+	if err == nil {
+		return moRef, nil
+	}
+
+	// Fallback to finding by inventory path
+	moRef, err = findVMWithInventoryPath(ctx)
+	if err != nil {
+		return moRef, err
+	}
+	return moRef, nil
+}
+
+// findVMWithBiosUUID searches for a VM by its BIOS UUID.
+// The 2nd return value indicates if the method was applicable to the VM context.
+func findVMWithBiosUUID(ctx *context.VMContext) (types.ManagedObjectReference, bool, error) {
```

Comment:
I am not in favor of returning another variable i.e `bool`. Let us find a way without this.

Reply:
Looking at it again, I agree. Will refactor to return an error instead!